grails-app/controllers/com/unifina/controller/api/StreamApiController.groovy

```diff
@@ -168,4 +172,26 @@ class StreamApiController {
 			action.call(stream)
 		}
 	}
+
+	@StreamrApi
+	def status() {
+		Stream s = Stream.get((String) params.id)
+		if (s == null) {
+			response.status = 404
```

Comment:
Created JIRA issue (CORE-1632) for myself to look into this response status code not being set by exceptions.